src/main/kotlin/uk/gov/dluhc/printapi/database/TwelveByteId.kt

```diff
@@ -0,0 +1,157 @@
+package uk.gov.dluhc.printapi.database
+
+import java.nio.ByteBuffer
+import java.security.SecureRandom
+import java.util.Date
+import java.util.concurrent.atomic.AtomicInteger
+
+/**
```

Comment:
IMHO 😁  this kdoc is better than the docs on mongo's `ObjectId` class. It explains the algorithm better I think.

As a side note it was actually quite good doing this work because it meant I had to understand how their class worked and understand the algorithm. It's really quite a neat approach, and is more than just "generate a random number / hex string" 👍

Reply:
It's also different to: https://www.mongodb.com/developer/products/mongodb/bson-data-types-objectid/